tests/beaker_tests/ntp_auth_key/ntp_auth_key_provider_defaults.rb

```diff
@@ -0,0 +1,129 @@
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+###############################################################################
+# TestCase Name:
+# -------------
+# NtpAuthKey-Provider-Defaults.rb
+#
+# TestCase Prerequisites:
+# -----------------------
+# This is a ntp_auth_key resource testcase for Puppet Agent on Nexus devices.
+# The test case assumes the following prerequisites are already satisfied:
+# A. Populating the HOSTS configuration file with the agent and master
+# information.
+# B. Enabling SSH connection prerequisites on the N9K switch based Agent.
+# C. Starting of Puppet master server on master.
+# D. Sending to and signing of Puppet agent certificate request on master.
+#
+# TestCase:
+# ---------
+# This is a ntp_auth_key resource test that tests for default value for
+# 'ensure' attribute of a ntp_auth_key resource.
+#
+# There are 2 sections to the testcase: Setup, group of teststeps.
+# The 1st step is the Setup teststep that cleans up the switch state.
+# Steps 2-4 deal with ntp_auth_key resource and its
+# verification using Puppet Agent and the switch running-config.
+#
+# The testcode checks for exit_codes from Puppet Agent, Vegas shell and
+# Bash shell command executions. For Vegas shell and Bash shell command
+# string executions, this is the exit_code convention:
+# 0 - successful command execution, > 0 - failed command execution.
+# For Puppet Agent command string executions, this is the exit_code convention:
+# 0 - no changes have occurred, 1 - errors have occurred,
+# 2 - changes have occurred, 4 - failures have occurred and
+# 6 - changes and failures have occurred.
+# 0 is the default exit_code checked in Beaker::DSL::Helpers::on() method.
+# The testcode also uses RegExp pattern matching on stdout or output IO
+# instance attributes of Result object from on() method invocation.
+#
+###############################################################################
+
+# Require UtilityLib.rb and NtpAuthKeyLib.rb paths.
+require File.expand_path('../../lib/utilitylib.rb', __FILE__)
+require File.expand_path('../ntp_auth_keylib.rb', __FILE__)
+
+result = 'PASS'
+testheader = 'ntp_auth_key Resource :: All Attributes Defaults'
+
+# @test_name [TestCase] Executes defaults testcase for ntp_auth_key Resource.
+test_name "TestCase :: #{testheader}" do
+  # @step [Step] Sets up switch for provider test.
+  step 'TestStep :: Setup switch for provider test' do
+    # Cleanup before starting test.
+    resource_absent_cleanup(agent, 'ntp_auth_key')
```

Comment:
We have had problems when tests bail out early so we have started using `teardown` so that it gets run on bailout.  Just create a cleanup method that calls `resource_absent_cleanup` and then call it in the setup section and inside the `teardown` method.

https://github.com/cisco/cisco-network-puppet-module/blob/2f1d60e358b7009bfedcf6175357603c9e115831/tests/beaker_tests/cisco_acl/test_ace.rb#L141